naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionField.scala

```diff
@@ -48,8 +50,13 @@ object NaptimeUnionField {
         case definitions: java.util.Map[String @unchecked, String @unchecked] => definitions.asScala
       }.getOrElse(Map[String, String]())
 
-      val unionMemberKey = typedDefinitions
-        .getOrElse(subType.getUnionMemberKey, subType.getUnionMemberKey)
+      val unionMemberKey = subType match {
+        case sType if typedDefinitions.contains(sType.getUnionMemberKey) =>
+          typedDefinitions(sType.getUnionMemberKey)
+        case sType: NamedDataSchema if typedDefinitions.contains(sType.getName) =>
```

Comment:
i also think this is only supposed to work if the union's namespace is the same as the child's namespace. can we check against that?